src/java/org/apache/cassandra/repair/autorepair/IAutoRepairTokenRangeSplitter.java

```diff
@@ -32,11 +37,25 @@ public interface IAutoRepairTokenRangeSplitter
      * The autorepair framework will repair the list of returned subrange in a sequence.
      * @param repairType The type of repair being executed
      * @param primaryRangeOnly Whether to repair only this node's primary ranges or all of its ranges.
-     * @param keyspaceName The keyspace being repaired
-     * @param tableNames The tables to repair
+     * @param keyspacesAndTablesToRepair A map keyed by keyspace name and valued by tables to generate assignments for.
      * @return repair assignments broken up by range, keyspace and tables.
      */
-    List<RepairAssignment> getRepairAssignments(AutoRepairConfig.RepairType repairType, boolean primaryRangeOnly, String keyspaceName, List<String> tableNames);
+    Map<String, List<RepairAssignment>> getRepairAssignments(AutoRepairConfig.RepairType repairType, boolean primaryRangeOnly, Map<String, List<String>> keyspacesAndTablesToRepair);
+
+    /**
+     * Reorders the list of {@link RepairAssignment} objects based on their priority for a given repair type.
+     * The list is sorted in descending order, so higher priority assignments appear first.
+     * If two assignments have the same priority for the specified repair type, their original order is preserved.
+     *
+     * @param repairAssignments A list of {@link RepairAssignment} objects to be reordered.
+     * @param repairType The {@link AutoRepairConfig.RepairType} used to determine the priority of each assignment.
+     *                   The priority is determined using the {@link RepairAssignment#getPriority(AutoRepairConfig.RepairType)} method.
+     */
+    @VisibleForTesting
+    default <T extends RepairAssignment> void reorderByPriority(List<T> repairAssignments, AutoRepairConfig.RepairType repairType)
+    {
+        repairAssignments.sort(Comparator.comparingInt(a -> ((RepairAssignment) a).getPriority(repairType)).reversed());
```

Comment:
I don't think there is any particular standard, but I could see people thinking lower number == higher priority.

 e.g. priority 1 should be the highest considered priority, with 2, 3, 4, 5, n being higher priority.   I think the main value add of that is by marking a table as priority 1, it should always be considered the most important to repair, where if you use higher numbers there needs to be an implicit agreed upon "highest priority".